lib/application/pages/home/collections/update/update_collection_page.dart

```diff
@@ -58,19 +60,46 @@ class _UpdateCollectionContents extends ConsumerWidget {
 
   @override
   Widget build(BuildContext context, WidgetRef ref) {
-    final vm = useUpdateCollectionVM(ref);
-    final state = useUpdateCollectionState(ref);
+    final vm = ref.read(updateCollectionVM.notifier);
+    final state = ref.watch(updateCollectionVM);
 
     if (state is UpdateCollectionFailedLoading) {
-      return Center(child: ExceptionRetryContainer(exception: state.exception, onRetry: vm.loadInitialContent));
+      return Center(child: ExceptionRetryContainer(exception: state.exception, onRetry: vm.loadContent));
     }
 
-    switch (selectedSegment) {
-      case _Segment.details:
-        return UpdateCollectionDetails();
-      case _Segment.memos:
-        return UpdateCollectionMemos();
+    if (state is UpdateCollectionLoading) {
+      return const Center(child: CircularProgressIndicator());
     }
+
+    if (state is UpdateCollectionLoaded) {
+      switch (selectedSegment) {
+        case _Segment.details:
+          return ProviderScope(
+            overrides: [
+              updateDetailsMetadata.overrideWithValue(state.collectionMetadata),
+            ],
+            child: _UpdateCollectionDetails(),
+          );
+        case _Segment.memos:
+          return UpdateCollectionMemos();
+      }
+    }
+
+    throw InconsistentStateError.layout('Unsupported subtype (${state.runtimeType}) of `UpdateCollectionState`');
+  }
+}
+
+class _UpdateCollectionDetails extends ConsumerWidget {
```

Comment:
Because `ref.watch(updateCollectionVM.notifier);` uses `updateDetailsMetadata` provider, which is injected in `UpdateCollectionDetails`. See SS below.

<details>
<summary>Screenshot</summary>

![CleanShot 2021-11-21 at 15 51 38](https://user-images.githubusercontent.com/11745745/142775136-11ed575f-addf-4e91-8974-14833de5e73d.png)


</details>

What I may do is to use `Consumer` to remove this wrapper Widget, like:

Before:
```dart
if (state is UpdateCollectionLoaded) {
      switch (selectedSegment) {
        case _Segment.details:
          return ProviderScope(
            overrides: [
              updateDetailsMetadata.overrideWithValue(state.collectionMetadata),
            ],
            child: _UpdateCollectionDetails(),
          );
        case _Segment.memos:
          return UpdateCollectionMemos();
      }
    }
```

After:
```dart
if (state is UpdateCollectionLoaded) {
      switch (selectedSegment) {
        case _Segment.details:
          return ProviderScope(
            overrides: [
              updateDetailsMetadata.overrideWithValue(state.collectionMetadata),
            ],
            child: Consumer(
              builder: (context, ref, child) {
                ref.listen<UpdatedDetailsState>(
                  updateCollectionDetailsVM,
                  (_, state) => vm.updateMetadata(metadata: state.metadata),
                );

                return UpdateCollectionDetails();
              },
            ),
          );
        case _Segment.memos:
          return UpdateCollectionMemos();
      }
    }
```

What do you think?

Reply:
Ahh got it. Made the changes.